kornia/feature/adalam/adalam.py

```diff
@@ -0,0 +1,230 @@
+        'scale_rate_threshold': 1.5,  # Maximum difference (ratio) in scales for a point to be accepted in a neighborhood. Set to None to disable the use of keypoint scales.   # noqa: E501
+        'detected_scale_rate_threshold': 5,  # Prior on maximum possible scale change detectable in image couples. Affinities with higher scale changes are regarded as outliers.   # noqa: E501
+        'refit': True,  # Whether to perform refitting at the end of the RANSACs. Generally improves accuracy at the cost of runtime.   # noqa: E501
+        'force_seed_mnn': True,  # Whether to consider only MNN for the purpose of selecting seeds. Generally improves accuracy at the cost of runtime.    # noqa: E501
+        # You can provide a MNN mask in input to skip MNN computation and still get the improvement.
+        'device': get_cuda_device_if_available(),  # Device to be used for running AdaLAM. Use GPU if available.   # noqa: E501
+    }
+    return DEFAULT_CONFIG
+
+
+def match_adalam(
+    desc1: Tensor,
+    desc2: Tensor,
+    lafs1: Tensor,
+    lafs2: Tensor,
+    config: Dict = get_adalam_default_config(),
+    hw1: Optional[Tensor] = None,
+    hw2: Optional[Tensor] = None,
+    dm: Optional[Tensor] = None,
+) -> Tuple[Tensor, Tensor]:
+    """Function, which performs descriptor matching, followed by AdaLAM filtering (see :cite:`AdaLAM2020` for more
+    details)
+
+    If the distance matrix dm is not provided, :py:func:`torch.cdist` is used.
+
+    Args:
+        desc1: Batch of descriptors of a shape :math:`(B1, D)`.
+        desc2: Batch of descriptors of a shape :math:`(B2, D)`.
+        lafs1: LAFs of a shape :math:`(1, B1, 2, 3)`.
+        lafs2: LAFs of a shape :math:`(1, B1, 2, 3)`.
+        config: dict with AdaLAM config
+        dm: Tensor containing the distances from each descriptor in desc1
+          to each descriptor in desc2, shape of :math:`(B1, B2)`.
+
+    Return:
+        - Descriptor distance of matching descriptors, shape of :math:`(B3, 1)`.
+        - Long tensor indexes of matching descriptors in desc1 and desc2. Shape: :math:`(B3, 2)`,
+          where 0 <= B3 <= B1.
+    """
+    KORNIA_CHECK_SHAPE(desc1, ["B", "DIM"])
+    KORNIA_CHECK_SHAPE(desc2, ["B", "DIM"])
+    KORNIA_CHECK_LAF(lafs1)
+    KORNIA_CHECK_LAF(lafs2)
+
+    adalam_object = AdalamFilter(config)
+    idxs = adalam_object.match_and_filter(
+        get_laf_center(lafs1).reshape(-1, 2),
+        get_laf_center(lafs2).reshape(-1, 2),
+        desc1,
+        desc2,
+        hw1,
+        hw2,
+        get_laf_orientation(lafs1).reshape(-1),
+        get_laf_orientation(lafs2).reshape(-1),
+        get_laf_scale(lafs1).reshape(-1),
+        get_laf_scale(lafs2).reshape(-1),
+    )
+    quality = torch.ones(len(idxs), 1, dtype=desc1.dtype, device=desc1.device)
```

Comment:
Because AdaLAM does not give confidence scores, it just removes those, which it think are wrong.

Reply:
Actually, the removal is based on thresholding a confidence score, so this can be used to estimate the keypoint quality! It's defined on line 304 of adalam/core.py under name conf, it is the ratio of the number of local inliers in a neighborhood divided by the number of expected false inliers for uniformly spread keypoints in the same neighborhood (so up to 1 the inliers are statistically insignificant, and the higher the better, unbounded). It's the same score that gets compared with the configuration "min_confidence" to decide the final filtering.

Reply:
Oh, cool, thank you!